netcdf4/src/main/java/ucar/nc2/jni/netcdf/Nc4Iosp.java

```diff
@@ -2339,8 +2350,10 @@ private void createGroup(Group4 g4) throws IOException {
 
     // dimensions
     for (Dimension dim : g4.g.getDimensions()) {
-      int dimid = addDimension(g4.grpid, dim.getShortName(), dim.getLength());
+      int dimLength = dim.isUnlimited() ? Nc4prototypes.NC_UNLIMITED : dim.getLength();
+      int dimid = addDimension(g4.grpid, dim.getShortName(), dimLength);
```

Comment:
Yep, and I originally created a fix that targeted the chunk sizes generated by `Nc4ChunkingDefault`. As it turns out, the chunk sizes were fine, but ONLY for an unlimited variable. And of course, we weren't marking the variable's dimension as unlimited.